inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -181,6 +179,48 @@ std::vector<std::shared_ptr<ngraph::op::util::SubGraphOp::OutputDescription>> V1
     return outputs;
 }
 
+ngraph::op::v5::Loop::SpecialBodyPorts V10Parser::XmlDeserializer::parsePurposeAttribute(const pugi::xml_node& node) {
+    ngraph::op::v5::Loop::SpecialBodyPorts result = {-1, -1};
+    std::map<uint64_t, uint64_t> params;
+    std::map<uint64_t, uint64_t> results;
+    map_type_in_function(node, "Parameter", params);
+    map_type_in_function(node, "Result", results);
+
+    NGRAPH_CHECK(!params.empty() || !results.empty(), "No parameters or results found in body Function.");
+
+    // Parse PortMap: external_port_id for inputs/outputs does not always appear in consecutive order
+    std::map<uint64_t, pugi::xml_node> input_map;
+    FOREACH_CHILD(_input, node.child("port_map"), "input") {
+        int64_t ext_port_id = GetInt64Attr(_input, "external_port_id");
+        input_map[ext_port_id] = _input;
```

Comment:
1) Consider `std::map::emplace`. 
Calling `std::map::[]` will create default value of `pugi::xml_node` and then assign `_intput` to it.
\* when switch to `emplace` behavior is slightly different. For `[]` wins last assigned and for `emplace` wins first.

2) IMHO we should avoid names which begins from underscore (just in case)
https://en.cppreference.com/w/cpp/language/identifiers#In_declarations

Reply:
Empleace is used instead std::map[]. underscore is removed.